src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts

```diff
@@ -2206,9 +2206,6 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 	}
 
 	public async getWorkspaceTasks(runSource: TaskRunSource = TaskRunSource.User): Promise<Map<string, IWorkspaceFolderTaskResult>> {
-		if (!(await this._trust())) {
-			return new Map();
-		}
 		await this._waitForSupportedExecutions;
```

Comment:
Trust is required before fetching tasks from extensions. It is very common for extension to run scripts or processes when they are creating/finding tasks. Because of this, we need to ask for trust before letting extensions find tasks.

Reply:
This was a deliberate decsion. I would strongly recommend against changing it as it would make it so task providing extensions run scripts even in untrusted workspaces.

Reply:
Wouldn't that only happen when they're requested to run though?

I'm not clear on the harm in registering providers or reading their task configurations, then requesting trust on attempt to run them.

Reply:
Simply enumerating tasks will cause us to go to each extension and call their [`provideTasks`](https://github.com/microsoft/vscode/blob/49f73faa18386aa79ee660fada24c9d3b332a3e0/src/vscode-dts/vscode.d.ts#L7624-L7625) method. `provideTasks` is often where extension will run scripts or otherwise execute code. For example, our built in gulp task provider does this by running gulp.